Fix kernel 4.10 rename of struct vm_fault->virtual_address to address

Snicksie opened the pull request:
Signed-off-by: Neline van Ginkel <neline.vanginkel@cs.kuleuven.be>

In kernel version 4.10 the field virtual_address in struct vm_fault was renamed to address.

lzha101 committed:
Merge pull request #17 from lzha101/master

Update License.txt to add Intel(R) before SGX

Signed-off-by: Lili Zhang <lili.z.zhang@intel.com>

guzongmin committed:
update the readme file

signed-off-by: Gu, Zongmin <zongmin.gu@intel.com>

lzha101 committed:
Merge pull request #18 from ZongminGu/master

Update the readme file

Signed-off-by: Lili Zhang <lili.z.zhang@intel.com>

Commit:
Fix kernel 4.10 rename of struct vm_fault->virtual_address to address

Signed-off-by: Neline van Ginkel <neline.vanginkel@cs.kuleuven.be>

haitaohuang commented:
This repo is intended to be out-of-tree driver to support the user space SDK and PSW before sgx driver upstreamed. Generally we would like patches submitted to the upstreaming tree first. Instructions are here: https://01.org/intel-software-guard-extensions/downloads/source-code-repositories.  We'll port the changes from upstreaming tree to this repo from time to time. The main purpose of this repo is to ensure the SGX PSW targeted versions of OS works. Based on that, I suggest you check and submit this to the upstreaming repo following review process in the mailing list describe in the page mentioned above.
thanks

Commit:
Bug 4093924 fix

BUG_ON macro call on a eremove failure was removed.
It caused an application freeze when eremove is triggered in
parallel with an enclave call.

Signed-off-by: Serge Ayoun <serge.ayoun@intel.co>
